src/linear_algebra/affine_space.lean

```diff
@@ -188,37 +198,174 @@ from the given points.  This is intended to be used when the sum of
+
+/-- The weighted sum, multiplied by a constant. -/
+lemma weighted_vsub_smul (r : k) (w : ι → k) (p : ι → P) :
+  r • s.weighted_vsub V w p = s.weighted_vsub V (r • w) p :=
+s.weighted_vsub_of_point_smul V r w p _
+
+/-- The weighted sum, negated. -/
+lemma weighted_vsub_neg (w : ι → k) (p : ι → P) :
+  -s.weighted_vsub V w p = s.weighted_vsub V (-w) p :=
+s.weighted_vsub_of_point_neg V w p _
+
+/-- Adding two weighted sums. -/
+lemma weighted_vsub_add (w₁ w₂ : ι → k) (p : ι → P) :
+  s.weighted_vsub V w₁ p + s.weighted_vsub V w₂ p = s.weighted_vsub V (w₁ + w₂) p :=
+s.weighted_vsub_of_point_add V w₁ w₂ p _
+
+/-- Subtracting two weighted sums. -/
+lemma weighted_vsub_sub (w₁ w₂ : ι → k) (p : ι → P) :
+  s.weighted_vsub V w₁ p - s.weighted_vsub V w₂ p = s.weighted_vsub V (w₁ - w₂) p :=
+s.weighted_vsub_of_point_sub V w₁ w₂ p _
+
+/-- A weighted sum of the results of subtracting a default base point
+from the given points, added to that base point.  This is intended to
+be used when the sum of the weights is 1, in which case it is an
+affine combination (barycenter) of the points with the given weights;
+that condition is specified as a hypothesis on those lemmas that
+require it. -/
+def affine_combination (w : ι → k) (p : ι → P) : P :=
+s.weighted_vsub_of_point V w p (classical.choice S.nonempty) +ᵥ (classical.choice S.nonempty)
+
+/-- `affine_combination` gives the sum with any base point, when the
+sum of the weights is 1. -/
+lemma affine_combination_eq_weighted_vsub_of_point_vadd_of_sum_eq_one (w : ι → k) (p : ι → P)
+    (h : ∑ i in s, w i = 1) (b : P) :
+  s.affine_combination V w p = s.weighted_vsub_of_point V w p b +ᵥ b :=
+s.weighted_vsub_of_point_vadd_eq_of_sum_eq_one V w p h _ _
+
+/-- Adding a `weighted_vsub` to an `affine_combination`. -/
+lemma weighted_vsub_vadd_affine_combination (w₁ w₂ : ι → k) (p : ι → P) :
+  s.weighted_vsub V w₁ p +ᵥ s.affine_combination V w₂ p = s.affine_combination V (w₁ + w₂) p :=
+begin
+  erw vadd_assoc,
+  congr,
+  exact s.weighted_vsub_add V w₁ w₂ p
+end
+
+/-- Subtracting two `affine_combination`s. -/
+lemma affine_combination_vsub (w₁ w₂ : ι → k) (p : ι → P) :
+  s.affine_combination V w₁ p -ᵥ s.affine_combination V w₂ p = s.weighted_vsub V (w₁ - w₂) p :=
+begin
+  erw vadd_vsub_vadd_cancel_right,
+  exact s.weighted_vsub_sub V w₁ w₂ p
+end
+
+end finset
+
+namespace affine_space
+
+open add_action
+open add_torsor
+
+variables {k : Type*} (V : Type*) {P : Type*} [ring k] [add_comm_group V] [module k V]
+variables [S : affine_space k V P] {ι : Type*} [fintype ι]
+include S
+
+/-- A weighted sum of the results of subtracting a base point from the
+given points.  The main cases of interest are where the sum of the
+weights is 0, in which case the sum is independent of the choice of
+base point, and where the sum of the weights is 1, in which case the
+sum added to the base point is independent of the choice of base
+point. -/
+def weighted_vsub_of_point (w : ι → k) (p : ι → P) (b : P) : V :=
```

Comment:
I'm not sure if this specialisation should get it's own definition, or whether we should just write `finset.univ` occasionally. It feels like you need to duplicate a fair amount of api this way.
At the moment everything is under control, but in the future, people will start proving lemmas about one of the two, and not duplicate them to the other...
It's not clear to me if this definition is really "worth" this duplication.

Reply:
Removed the duplicate versions, so users for a `fintype` that need to name the `finset` explicitly will need to use `finset.univ` (but some users may well just be able to rewrite without needing to name the `finset`).